Lib/test/test_xml_etree.py

```diff
@@ -692,28 +692,31 @@ def test_iterparse(self):
         it = iterparse(TESTFN)
         action, elem = next(it)
         self.assertEqual((action, elem.tag), ('end', 'document'))
-        with warnings_helper.check_no_resource_warning(self):
-            with self.assertRaises(ET.ParseError) as cm:
-                next(it)
-            self.assertEqual(str(cm.exception),
-                    'junk after document element: line 1, column 12')
-            del cm, it
+        with self.assertRaises(ET.ParseError) as cm:
```

Comment:
This seems like it should still be validating no resource warning; just add the close to make sure other resource warnings don't accidentally sneak in